### PR TITLE
fix Hide video playback controls on auto-playing videos

### PR DESCRIPTION
**Pull request by tienifr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/42323
PROPOSAL: https://github.com/Expensify/App/issues/42323#issuecomment-2116490326

### Tests

1. On a new account, tap green (+) button
2. Tap "Track Expense"
3. Observe bottom sheet w/video about how to Track expense playing with controls does not display by default
- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

**Prerequisite**: Native and mWeb platforms only

A) **Testing feature training video player**:
1. On a new account, tap green (+) button
2. Tap "Track Expense"
3. Verify that the training video for `Track Expense` does not show the video controls by default. (#Test 1)
4. Tap inside the `video player` once (i.e. do not tap repeatedly) while the video still plays.
5. Verify that the `video controls` are displayed in the `video player`(#Test 2)
6. Also verify that the video playing status (i.e. play/pause) does not change (#Test 3)
7. Also verify that the `video controls` fades away after 1.5 seconds (#Test 4)
8. Tap inside the `video player` again so that the `video controls` are displayed.
9. Tap again inside the `video player` within 1.5 seconds and verify that the `video controls` fade away (#Test 5)
10. Tap inside the `video player` once to show the `video controls` again.
11. Tap on the `Pause` button available in the `video controls` to pause the play and verify that the `video controls` do not fade away (#Test 6)
12. Tap on the `Play` button available in the `video controls` to start playing again
13. While playing, tap again on `3DotMenu` available in the `video controls` and verify that the `video controls ` do not fade away (#Test 7)
14. Tap on the `Got it` button to close the feature training video.

B) **Testing non-feature training video player**:
1. Navigate to Concierge Chat. 
2. Tap on the `attachment` to launch the video which explains how to track your spend
3. Verify that the displayed `video controls` fade away after 1.5 seconds. (#Test 8)
4. Repeat steps (A-6) to (A-12) as mentioned above.

**Prerequisite**: Web and Desktop versions.
C) **Testing non-feature and feature training video player**
1. On a new account, tap green (+) button
2. Tap "Track Expense"
3. Verify that the training video for `Track Expense` does not show the video controls by default. (#Test 9)
4. Hover inside the video player and verify that video controls are displayed (#Test 10)
5. Other than the above step, all other video player features remain the same without any change.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/113963320/06d5f853-54d7-4538-8fa2-da828a822e9e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/113963320/3de4a577-4cf6-415c-bbff-8b7b932d5aa7
</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/113963320/a793a107-23d0-40d4-8ea3-4a70117e57c3

https://github.com/user-attachments/assets/36758689-d882-42ec-9204-f25088853e69

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by dubielzyk-expensify** [comment]
The video on iOS don't seem to be autoplaying is that correct?

Can you do some recordings of how this looks for a in a chat thread?

Checked out the PR locally and the desktop versions feels good to me. A bit harder to get a feel for native, but it looks alright at first glance.

cc @Expensify/design

**Comment by dubielzyk-expensify** [comment]
Going OOO for the next 7 days, so @shawnborton and @dannymcclain will be able to assist with the last few details :)

**Comment by rojiphil** [comment]
Reviewing today

**Review by rojiphil — COMMENTED** [review]
For feature training videos on mobile devices, the video plays automatically and we do not show the controls by default. This works fine. 
1. But when a video `Attachment` is played (I.e. either via tap or on the `play` button on `playback controls`), the `playback controls` do not fade away as per OP expectations. Did we miss implementing this?

Also, I have a few additional observations and follow-up queries:

2. Currently, the fading away feature will get triggered when `played` and `paused`. 
Curious to know if the `playback controls` should remain shown when `paused` on mobile. Or is it fine for the `playback controls` to fade away when `paused`?

3. Also, when popover is visible, shouldn’t the `playback controls` remain displayed and not fade away?

What do you think?

**Comment by rojiphil** [comment]
> Can you do some recordings of how this looks for a in a chat thread?

Please find recordings below. 
I have also left some follow-up queries [here](https://github.com/Expensify/App/pull/43349#pullrequestreview-2117539374)

<details>
<summary>iOS Native</summary>

https://github.com/Expensify/App/assets/3004659/e9cdd8f0-eef2-43be-a4ad-fb7fc301f9b4

</details>

<details>
<summary>mWeb Chrome</summary>

https://github.com/Expensify/App/assets/3004659/c2cd84ab-c713-4a95-a001-914b79559119

</details>

**Comment by shawnborton** [comment]
> But when a video Attachment is played (I.e. either via tap or on the play button on playback controls), the playback controls do not fade away as per OP expectations. Did we miss implementing this?

Perhaps we did not implement this, and we probably should implement it? 

> Currently, the fading away feature will get triggered when played and paused. Curious to know if the playback controls should remain shown when paused on mobile. Or is it fine for the playback controls to fade away when paused?

I think I would expect the controls to remain there even if the video is paused? But if you tap the video outside of the pause button, I would expect the video to remain paused but the controls to fade away. 

> Also, when popover is visible, shouldn’t the playback controls remain displayed and not fade away?

Popover as in the training modal? I think we'd want to fade those away too.

**Comment by rojiphil** [comment]
> > Also, when popover is visible, shouldn’t the playback controls remain displayed and not fade away?
> 
> Popover as in the training modal? 

I meant the popover menu with menu options `Download` and `Playback speed` as shown below. Suppose the user makes the popover menu visible while the `playback controls` are shown (i.e. before it fades away). In that case, the menu options remain displayed even after the `playback controls` fade. 
 
Screenshot 2024-06-14 at 12 07 38 PM

**Comment by rojiphil** [comment]
> > But when a video Attachment is played (I.e. either via tap or on the play button on playback controls), the playback controls do not fade away as per OP expectations. Did we miss implementing this?
> 
> Perhaps we did not implement this, and we probably should implement it?

> > Currently, the fading away feature will get triggered when played and paused. Curious to know if the playback controls should remain shown when paused on mobile. Or is it fine for the playback controls to fade away when paused?
> 
> I think I would expect the controls to remain there even if the video is paused? But if you tap the video outside of the pause button, I would expect the video to remain paused but the controls to fade away.

So, if I understand correctly, 
On mobile devices,
1. When the video is tapped outside of the `playback controls` (whether `play` or `pause`), we show the `playback controls` and fade them after the agreed interval.
2. When the `playback controls` are interacted upon by clicking any of the button, we keep showing the `playback controls` without fading them.
3.  Video `attachment` also behaves as mentioned in 1. and 2.  But, unlike `feature training` videos, the `playback controls` is shown initially by default and fades away.  

Is this understanding correct?

**Comment by shawnborton** [comment]
I think that sounds right to me, yup.

**Comment by shawnborton** [comment]
> Therefore, fixing the attachment video would expand the scope of this issue. What are your thoughts?

This seems okay to me though. Or maybe, what exactly are you asking me here? Anything else you need me to clear up?

**Comment by shawnborton** [comment]
Ah, I think maybe the idea is that as long as the video is playing, the controls will eventually fade out. Then you tap to get them to come back. But if you tap and pause, the controls stay in place. Does that sound right @Expensify/design ?

**Comment by shawnborton** [comment]
Going to create an adhoc build for testing

**Comment by shawnborton** [comment]
> Then, tap on the video. The video is paused, the control is displayed without fading.

Hmm, I'm not sure if we want tapping on the video to pause it. I would think tapping on the video just shows the controls. Thoughts @Expensify/design ?

**Comment by shawnborton** [comment]
So I think we're close, but I think tapping on the video shouldn't pause or play it. Only tapping on the pause/play icon would do that.

**Comment by dannymcclain** [comment]
>Hmm, I'm not sure if we want tapping on the video to pause it. I would think tapping on the video just shows the controls.

Yeah I agree with you Shawn.

**Comment by shawnborton** [comment]
Understood, I suggest that we change that though to what we are all agreeing on above.

**Comment by shawnborton** [comment]
Yeah I think that's a fair concern. It seems like this pattern of tapping the video to show/hide controls is mostly just for mobile. So I would think on desktop, we would keep the behavior of tapping anywhere to play/pause.

**Comment by shawnborton** [comment]
That sounds right to me, mind sharing a screen recording too?

**Comment by shawnborton** [comment]
Awesome, that looks pretty good to me!

File: src/components/VideoPlayer/BaseVideoPlayer.tsx
Comment (rojiphil) [inline src/components/VideoPlayer/BaseVideoPlayer.tsx:355]: NAB but can we prevent calling `showControl` for large screen devices? Maybe via an early return after calling `togglePlayCurrentVideo`. It will also keep possible future regressions at bay.

**Review by rojiphil — COMMENTED** [review]
@tienifr Thanks for the PR. Overall the feature works well but I have few observations as described below:
1. In step (3) of comment [here](https://github.com/Expensify/App/pull/43349#issuecomment-2194142166), shouldn’t the show duration timer be reset again so that the playback controls would be shown for 2 sec from the `latest` tap on video? Currently, it does not reset and the controls are hidden after 2 sec duration of the `first` tap on video.
2. When `play` is going on for feature training videos and the popover menu is displayed, I think the `controls` should stay visible. Does this make sense?
3. For non feature training videos, the playback controls are not hidden after 2 seconds. Did we agree to leave this as it is?

Here is a video demonstrating the above.

https://github.com/Expensify/App/assets/3004659/2dfc7bc2-72e1-47f5-9fe2-3ceb2090ae68

**Comment by shawnborton** [comment]
> I believe this falls outside the scope of the current issue. Our current focus is on the feature training video, where we aim to hide the controls while still allowing user interaction if necessary. Extending this behavior to other elements like attachments or receipts requires further discussion regarding their specific behavior. Therefore, I suggest creating a separate issue for those cases. cc @shawnborton, your thoughts?

Hmm I disagree with that. The behaviors we are implementing here should be used globally for all video playback on mobile devices.

**Comment by rojiphil** [comment]
> > 2\. When play is going on for feature training videos and the popover menu is displayed, I think the controls should stay visible. Does this make sense?
> 
> * I don't think so. Our expected behavior in this issue is "The controls on the video shouldn't have the controls showing by default".

Hmm.. Not quite sure but may be and am fine with it also. But, let us check with Design to be doubly sure.
@shawnborton Can you please confirm the expected behavior on point (2)? 
The video [here](https://github.com/Expensify/App/pull/43349#pullrequestreview-2150598087) from around 19 sec onwards demonstrates this.

**Comment by dubielzyk-expensify** [comment]
In terms of functionality I agree with everything on what's done so far.

I think the playback controls should show when the 3 dots are pressed because if not, I assume desktop would feel weird with the playback controls not showing underneath but the popover still shows.

**Comment by shawnborton** [comment]
> 2. When play is going on for feature training videos and the popover menu is displayed, I think the controls should stay visible. Does this make sense?

Yeah I agree that in this case, where there is an active popover situation, we want the popover & controls to continue showing even after a few seconds pass.

**Comment by shawnborton** [comment]
Nice, that feels pretty good to me!

**Comment by dubielzyk-expensify** [comment]
Looks great!

**Comment by rojiphil** [comment]
@tienifr Can you please merge with the latest main? Will take this up for re-review today.

**Comment by rojiphil** [comment]
@tienifr Thanks for the merge. (1) and (2) as mentioned [here](https://github.com/Expensify/App/pull/43349#pullrequestreview-2150598087) is fixed but I think we missed implementing (3). For mobile devices, since we are hiding the controls after 2 sec for feature training videos, it would look awkward for non feature training videos to keep showing the controls without hiding after 2 sec. 
Can we please also get this done and we are through with this issue? Thanks

Also, there is a lint error that needs to be addressed.

> > 3\. For non feature training videos, the playback controls are not hidden after 2 seconds. Did we agree to leave this as it is?
> 
> * I believe this falls outside the scope of the current issue. Our current focus is on the feature training video, where we aim to hide the controls while still allowing user interaction if necessary. Extending this behavior to other elements like attachments or receipts requires further discussion regarding their specific behavior. Therefore, I suggest creating a separate issue for those cases. cc @shawnborton, your thoughts?

> 
> Hmm I disagree with that. The behaviors we are implementing here should be used globally for all video playback on mobile devices.

**Comment by rojiphil** [comment]
@tienifr Thanks for the changes. The mobile devices works awesome. But I think we have broken the web/desktop versions as the `video controls` fades away here. Isn’t the fading away UI experience only for mobile devices?
The test video [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235701370) demonstrates this.

**Comment by rojiphil** [comment]
@shawnborton Just wanted to confirm if it is expected for the `video controls` to be initially hidden for feature training videos on large screen devices as mentioned in step C-3 [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235701370)

**Comment by shawnborton** [comment]
I kind of think that if a video autoplays on desktop, then we wouldn't show the controls by default until the video is hovered. Thoughts on that @Expensify/design ?

**Comment by dubielzyk-expensify** [comment]
> @shawnborton Just wanted to confirm if it is expected for the video controls to be initially hidden for feature training videos on large screen devices as mentioned in step C-3 https://github.com/Expensify/App/pull/43349#issuecomment-2235701370

Not Shawn, but I think that's what we should do. Happy to have @shawnborton confirm/deny :)

**Comment by dubielzyk-expensify** [comment]
Jinx. Agreed Shawn.

**Comment by rojiphil** [comment]
> I kind of think that if a video autoplays on desktop, then we wouldn't show the controls by default until the video is hovered.

Thanks. So, that would mean C-4 steps would be:
4	Hover inside the `video player` and verify that `video controls` are displayed (#Test 10)

**Comment by rojiphil** [comment]
@tienifr This means we have only two remaining issues to nail this down. 1) [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235702333) and 2) [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235731139)

**Comment by dubielzyk-expensify** [comment]
In your mobile video it looks like it's correct where the controls aren't showing, but given your cursor is above the video when it pops up it's hard to tell. Can you do a recording where you tap track submit and your cursor goes off to see if the controls never show?

**Review by rojiphil — COMMENTED** [review]
@tienifr I have left a comment for an issue that is demonstrated in test video below. Please have a look. Otherwise, I think we are very close to getting this done.

https://github.com/user-attachments/assets/40ea31fe-4ac5-4de1-96f9-f97effa8a329

**Comment by rojiphil** [comment]
> In your mobile video it looks like it's correct where the controls aren't showing, but given your cursor is above the video when it pops up it's hard to tell. Can you do a recording where you tap track submit and your cursor goes off to see if the controls never show?

@dubielzyk-expensify From the code perspective, there is only one pending issue left. As soon as it is addressed, I suggest that we trigger a build (via `Ready to Build`) so that @Expensify/Design can have the final review on ux before we wrap-up on this.

**Comment by rojiphil** [comment]
Yeah. Reviewing in parallel maybe even better. Just one thing to note. There is one pending issue as mentioned [here](https://github.com/Expensify/App/pull/43349#pullrequestreview-2201186408)

**Comment by dubielzyk-expensify** [comment]
Ohh this is looking good. I got two notes:
1. Can we made the delay before the controls fade starts slightly shorter? What's it at the moment? Felt like is was more than 2 seconds.
2. It would be awesome if we could get the controls to disappear when you tap on the video as well (when the controls are showing). The auto fade is mostly for when you don't do anything and to make sure the controls disappear automatically when the video is playing.

**Comment by dubielzyk-expensify** [comment]
> Currently, the controls stay visible fully for 2000ms, and fade out for 500ms. I made it disappear at once.

I think keep everything as it was, just reduce the 2000ms down to 1500ms.

**Comment by rojiphil** [comment]
> 2\. It would be awesome if we could get the controls to disappear when you tap on the video as well (when the controls are showing).

@dubielzyk-expensify The commits after the above statements has impacted test step A-9 as mentioned [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235701370). As per step A-9, if we keep continually tapping on the video player within the timeout interval, the video controls remain shown. However, the controls will auto hide if timeout interval is exhausted after the most recent tap inside video player. The recent commits though forcefully hides the controls on every tap thereby impacting step A-9. Here is a test video after recent commits and I think it’s jittery. Can you confirm if it was intentional to change step A-9 or were you just concerned about reducing the time interval from 2 sec to 1.5 sec?


https://github.com/user-attachments/assets/6bedb3c6-6aa3-434e-a6f5-06992f09adc1

**Comment by dubielzyk-expensify** [comment]
> The recent commits though forcefully hides the controls on every tap thereby impacting step A-9.

The goal was that if the controls aren't showing, tapping the screen should show them. And reversely, if the controls aren't showing, tapping them show reveal them. So tapping many times on the screen should show, hide, repeat, multiple times. Tapping the screen should reset the timer for auto-hiding the controls so if a user interacts with it, it should override that behaviour anyways. 

Basically the list of interactions are as follow:
1. Auto-hide controls on startup
2. Tapping the video canvas should reveal the controls
3. If nothing is tapped again, then the controls should fade away after 1.5 seconds.
4. But if a user taps the video canvas again with the controls showing, we should hide the controls

Does that make sense?

**Comment by rojiphil** [comment]
Thanks @dubielzyk-expensify for the clarification.
Accordingly, Step A-9 is changed from
`9. Tap again inside the video player repeatedly within 2 seconds and verify that the video controls do not fade away (#Test 5)`
to the following
`9. Tap again inside the `video player` within 1.5 seconds and verify that the `video controls` fade away (#Test 5)`

**Comment by rojiphil** [comment]
@tienifr Can you please merge with the latest main? And I can complete the checklist. Thanks.

**Comment by rojiphil** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/1f49c363-4206-437d-ab45-cb7fefcca8a2

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/bf71c022-056b-40f0-906c-66355bf394ce

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/580af561-aadf-4d57-a736-c4c531a512c2

</details>

<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/b13bc0e8-6366-476d-aae5-b33b5137cd02

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/b0ed8b7c-f37b-418e-b757-38f28a6a432b

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/4e321833-8b90-45fc-a360-b86231a86d35

</details>

**Review by rojiphil — APPROVED** [review]
@tienifr Thanks. Just one note. Please update the test cases for QA as mentioned  [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235701370).

@dubielzyk-expensify Can you please also have one final look at the test steps [here](https://github.com/Expensify/App/pull/43349#issuecomment-2235701370) just to ensure that we are all on the same page? Thanks.

@MariaHCD I have completed the reviewer checklist and the improvements works well too. Over to you for review.

**Review by MariaHCD — APPROVED** [review]
LGTM, after approval from design, we should be good to go

**Comment by dubielzyk-expensify** [comment]
Those test steps look good to me 👍

**Comment by MariaHCD** [comment]
@tienifr @rojiphil let's update the test steps for QA in the OP?

**Comment by rojiphil** [comment]
> @tienifr @rojiphil let's update the test steps for QA in the OP?

Oh! I do not have privileges to update the OP. 
@tienifr Can you please update the QA steps as mentioned [here](https://github.com/Expensify/App/pull/43349#pullrequestreview-2218203030)? Thanks

**Comment by Beamanator** [comment]
FYI I believe this was deployed to prod yesterday, from this checklist - https://github.com/Expensify/App/issues/47219

**Pull request merged**
